File: src/04-challenge/solution.ts
export default function createXmasTree(height: number, ornament: string): string {
	let xmasTree: string = ''

	let i: number
	let branchBlankSpaces: string

	const trunkBlankSpaces: string = '_'.repeat(height - 1)

	for (i = 0; i < height; i++) {
		branchBlankSpaces = '_'.repeat(height - i - 1)
		xmasTree += `${branchBlankSpaces}${ornament.repeat(i * 2 + 1)}${branchBlankSpaces}\n`
	}

	xmasTree += `${trunkBlankSpaces}#${trunkBlankSpaces}\n`
	xmasTree += `${trunkBlankSpaces}#${trunkBlankSpaces}`

	return xmasTree
}

File: src/04-challenge/solution.py
def createXmasTree(height: int, ornament: str) -> str:
    tree: str = ""

    trunk_blank_spaces: str = "_" * (height - 1)

    for i in range(height):
        branch_blank_spaces: str = "_" * (height - i - 1)
        tree += (
            branch_blank_spaces + ornament * (i * 2 + 1) + branch_blank_spaces + "\n"
        )

    tree += trunk_blank_spaces + "#" + trunk_blank_spaces + "\n"
    tree += trunk_blank_spaces + "#" + trunk_blank_spaces

    return tree
